code-submissions-clustering-core/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/load/visualization/SubmissionsGraphVisualizer.kt

```diff
@@ -0,0 +1,99 @@
+package org.jetbrains.research.code.submissions.clustering.load.visualization
+
+import org.jetbrains.research.code.submissions.clustering.model.SubmissionsGraph
+import org.jetbrains.research.code.submissions.clustering.model.SubmissionsGraphAlias
+import org.jetbrains.research.code.submissions.clustering.model.SubmissionsNode
+import org.jetbrains.research.code.submissions.clustering.util.addFileToProject
+import org.jetbrains.research.code.submissions.clustering.util.deleteFromProject
+import org.jetbrains.research.code.submissions.clustering.util.getTmpProjectDir
+import org.jgrapht.alg.interfaces.ClusteringAlgorithm.Clustering
+import java.io.File
+
+class SubmissionsGraphVisualizer {
+    fun visualize(submissionsGraph: SubmissionsGraph, outputFile: File) {
+        val basePath = getTmpProjectDir(toCreateFolder = false)
+        val fileName = "graph.dot"
+        val dotRepresentation = toDot(submissionsGraph)
+        val dotFile = addFileToProject(basePath, fileName, dotRepresentation)
+        val cmd = "dot -Tpng $dotFile -o $outputFile"
+        val process = Runtime.getRuntime().exec(cmd)
+        process.waitFor()
+        dotFile.deleteFromProject()
+    }
+
+    fun toDot(submissionsGraph: SubmissionsGraph): String {
+        val stepId = submissionsGraph.graph.vertexSet().firstOrNull()?.stepId ?: "?"
+        var dotRepresentation = "graph $stepId {\n"
```

Comment:
System line separator instaed of \n?

Reply:
Also, please use the string builder here and in the other functions, because it is not efficient to plus strings